viite-UI/src/model/ProjectCollection.js

```diff
@@ -269,34 +271,37 @@
     var createOrUpdate = function (dataJson) {
       if ((!_.isEmpty(dataJson.linkIds) || !_.isEmpty(dataJson.ids)) && typeof dataJson.projectId !== 'undefined' && dataJson.projectId !== 0) {
         if (dataJson.roadNumber !== 0 && dataJson.roadPartNumber !== 0) {
-        resetEditedDistance();
-        var ids = dataJson.ids;
-            if (dataJson.linkStatus === LinkStatus.New.value && ids.length === 0) {
-              backend.createProjectLinks(dataJson, function (successObject) {
-                if (!successObject.success) {
-                  new ModalConfirm(successObject.errorMessage);
-                  applicationModel.removeSpinner();
-                } else {
-                  publishableProject = successObject.publishable;
-                  projectErrors = successObject.projectErrors;
-                  eventbus.trigger('projectLink:projectLinksCreateSuccess');
-                  eventbus.trigger('roadAddress:projectLinksUpdated', successObject);
-                }
-              });
-            } else {
-              backend.updateProjectLinks(dataJson, function (successObject) {
-                if (successObject.success) {
-                  publishableProject = successObject.publishable;
-                  projectErrors = successObject.projectErrors;
-                  eventbus.trigger('roadAddress:projectLinksUpdated', successObject);
-                } else {
-                  new ModalConfirm(successObject.errorMessage);
-                  applicationModel.removeSpinner();
-                }
-              });
-            }
+          applicationModel.addSpinner();
+          resetEditedDistance();
+          var ids = dataJson.ids;
+          if (dataJson.linkStatus === LinkStatus.New.value && ids.length === 0) {
+            backend.createProjectLinks(dataJson, function (successObject) {
+              if (!successObject.success) {
+                new ModalConfirm(successObject.errorMessage);
+                applicationModel.removeSpinner();
+              } else {
+                publishableProject = successObject.publishable;
+                projectErrors = successObject.projectErrors;
```

Comment:
same as previous: we can start to use the existing methods setProjectErrors, setReservedParts and setFormedParts

Reply:
For this _Numerointi_ option to be hidden when is not possible: I've matched the road number and road part number with the ones already modified (formed list):
 * If the road part and road part number is present in the modified roads (formed list), then the option _Numerointi_ is not possible;

Here are an explanation of the bugs found and what was fixed:
 1. When transferring a link, the _Numerointi_ option is still appearing on the dropdown:
 * **Reason**: The reason this was happening it's because the _formed list_ wasn't being updated.
 * **What was done**: I added an update after the user does any change in project links - so the _formed list_ keeps synchronized with the database.
 * **Current situation**: This update fixed the synchronization of the _formed list_ when any change is made. Now the _Numerointi_ option is being hidden when it should.

2. When reverting the renumbering, the _Numerointi_ option is still hidden:
 * **Reason*: As you know, the spin is added whenever the data is being loaded, the problem here is that the spin was beeing removed sooner than expectded.
 * **What was done**: Delayed when the spin is removed.
 * **Current situation**: I hope this changes are good enough - I still need this to be tested in devtest since the environment is faster and we can see some different behaviors happening.

3. When reverting any other change, the Numerointi option is still hidden:
 * *Reason*: The reason this was happening it's because the _formed list_ wasn't being updated.
 * *What was done*: I added an update after the user does any revert change in project links - so the _formed list_ keeps synchronized with the data base.
 * **Current situation** (similar with point **1.**): This update fixed the synchronization of the _formed list_ when any change is reverted. Now the _Numerointi_ option is visible as it should.